crunchers/cruncher.h

```diff
@@ -302,32 +311,39 @@ class StreamedMultiCruncher : public MultiCruncher<CRUNCHER> {
   virtual ~StreamedMultiCruncher() = default;
 
   void OnTick(std::chrono::microseconds us) {
-    std::chrono::microseconds last_us = last_event_us_;
-    for (size_t i = 0, sz = streams_.size(); i < sz; ++i) {
-      const uint64_t publish_interval = params_[i].interval.count();
-      const uint64_t start_marker = last_us.count() / publish_interval;
-      const uint64_t end_marker = us.count() / publish_interval;
-      for (uint64_t j = start_marker; j < end_marker; ++j) {
-        response_t response;
-        response.timestamp = std::chrono::microseconds((j + 1) * publish_interval);
-        crunchers_[i]->OnTick(response.timestamp);
-        response.value = crunchers_[i]->GetValue();
-        streams_[i].Publish(response);
-      }
-    }
+    OutputCrunchersDataUntil(us);
     base_t::OnTick(us);
   }
 
   void OnEvent(event_t&& event, idxts_t idxts) {
-    OnTick(idxts.us);
+    OutputCrunchersDataUntil(idxts.us);
     base_t::OnEvent(std::move(event), idxts);
   }
 
   void OnRequest(Request&& r) { base_t::OnRequest(std::move(r)); }
 
  private:
+  void OutputCrunchersDataUntil(std::chrono::microseconds us) {
+    const std::chrono::microseconds last_us = last_event_or_tick_us_;
+    for (size_t i = 0, sz = streams_.size(); i < sz; ++i) {
+      const auto publish_interval = params_[i].interval;
+      for (auto t = std::chrono::microseconds((last_us.count() / publish_interval.count() + 1) *
+                                              publish_interval.count());
+           t < us;
+           t += publish_interval) {
+        // The result of GetValue() call could be different if corresponding cruncher
```

Comment:
The result of  **the** ... if **the** corresponding cruncher ... that **the** "actual" timestamp has changed. :)